diagnostics/collector.go

```diff
@@ -0,0 +1,170 @@
+		"pg_stat_database",
+	}
+	for _, tbl := range contentFromTables {
+		err = c.writeTableContent(ctx, writer, tbl)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("write table content for %s %w", tbl, err))
+		}
+	}
+
+	countFromTables := []string{
+		"catalog_branches",
+		"catalog_commits",
+		"catalog_repositories",
+		"auth_users",
+	}
+	for _, tbl := range countFromTables {
+		err = c.writeTableCount(ctx, writer, tbl)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("write table count for %s %w", tbl, err))
+		}
+	}
+
+	err = c.writeRawQueryContent(ctx, writer, "table_sizes", `
+SELECT *, pg_size_pretty(total_bytes) AS total
+    , pg_size_pretty(index_bytes) AS INDEX
+    , pg_size_pretty(toast_bytes) AS toast
+    , pg_size_pretty(table_bytes) AS TABLE
+  FROM (SELECT *, total_bytes-index_bytes-COALESCE(toast_bytes,0) AS table_bytes FROM (
+      SELECT c.oid,nspname AS table_schema, relname AS TABLE_NAME
+              , c.reltuples AS row_estimate
+              , pg_total_relation_size(c.oid) AS total_bytes
+              , pg_indexes_size(c.oid) AS index_bytes
+              , pg_total_relation_size(reltoastrelid) AS toast_bytes
+          FROM pg_class c
+          LEFT JOIN pg_namespace n ON n.oid = c.relnamespace
+          WHERE relkind = 'r') a) a`)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("get table sizes %w", err))
+	}
+
+	// write all errors into log
+	if err := c.writeErrors(ctx, writer, errs); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Collector) writeQueryContent(ctx context.Context, writer *zip.Writer, name string, selectBuilder sq.SelectBuilder) error {
+	q := selectBuilder.Limit(maxRecordsPerQueryCollect)
+	query, args, err := q.ToSql()
+	if err != nil {
+		return fmt.Errorf("query build: %w", err)
+	}
+	return c.writeRawQueryContent(ctx, writer, name, query, args...)
+}
+
+func (c *Collector) writeRawQueryContent(ctx context.Context, writer *zip.Writer, name string, query string, args ...interface{}) error {
+	rows, err := c.db.Query(query, args...)
+	if err != nil {
+		return fmt.Errorf("execute query: %w", err)
+	}
+	defer rows.Close()
+
+	filename := name + csvFileExt
+	w, err := writer.Create(filename)
+	if err != nil {
+		return fmt.Errorf("new file for table %s - %w", name, err)
+	}
+	csvWriter := csv.NewWriter(w)
+	defer csvWriter.Flush()
+
+	first := true
+	for rows.Next() {
+		if err := ctx.Err(); err != nil {
+			return err
+		}
+		if first {
+			first = false
+			descriptions := rows.FieldDescriptions()
+			if len(descriptions) == 0 {
+				return ErrNoColumnsFound
+			}
+			cols := make([]string, len(descriptions))
+			for i, fd := range descriptions {
+				cols[i] = string(fd.Name)
+			}
+			if err := csvWriter.Write(cols); err != nil {
+				return fmt.Errorf("write csv header for %s %w", name, err)
+			}
+		}
+		v, err := rows.Values()
+		if err != nil {
+			return err
+		}
+		record := make([]string, len(v))
+		for i := range v {
+			record[i] = fmt.Sprintf("%v", v[i])
```

Comment:
Is this csv formatted? Don't you need a `","` between the columns of each row?

Reply:
The csv writer does the CSV formatting -- it will also handle value escape like using ',' in the value.